xmipp

```diff
@@ -698,6 +698,95 @@ def runTests(testNames):
     testsToDo = '--all' if len(testNames)==0 else ' '.join(testNames)
     runJob("cd %s; %s base.py %s" % (testsPath, python, testsToDo))
 
+def readModelsFile():
+    with open(os.path.join("src", "xmipp", "models", "models.list"), 'r') as f:
+        lines = f.readlines()
+        return [line.strip('\n') for line in lines if not line.startswith('#')]
+
+def addDeepLearninModel(modelPath, update=False):
+    """ Takes the folder from modelPath and makes a .tgz,
+        uploads the .tgz to scipion web and
+        adds the modelName to the model.list.
+    """
+    if not os.path.isdir(modelPath):
+        print("modelPathDir must be a folder. \n"
+              "The name of the model will be that folder name.\n")
+        usage()
+        sys.exit(1)
+
+    modelName = os.path.basename(modelPath)
+    modelList = readModelsFile()
+    if modelName in modelList:
+        print(red("'%s' model name is already used. " % modelName))
+        if update:
+            print("Are you sure that you want to OVERWRITE that model: (yes/NO)")
+            yesno = raw_input()
+            if yesno.lower() != 'yes':
+                sys.exit(1)
+        else:
+            print("Please, choose one skipping the following list:")
+            for line in modelList:
+                print(' - ' + line)
+            print(blue("If you want to overwrite the existing model "
+                       "add the '--update' option."))
+            sys.exit(1)
+
+    print("You will add a new model with the name: %s" % modelName)
+    print("Do you agree? (YES/no)")
+    yesno = raw_input()
+    if yesno.lower() == 'no':
+        print("The name is took from the folder name.")
+        sys.exit(1)
+
+    dirPath = os.path.dirname(modelPath)
+    print("Creating the '%s.tgz' model." % modelName)
+    runJob("cd %s ; tar czf %s.tgz %s ; cd - ; mv %s/%s.tgz %s.tgz"
+           % (dirPath, modelName, modelName, dirPath, modelName, modelName))
+
+    print("Trying to upload the model:")
+    login = "scipion@scipion.cnb.csic.es"
+    depositionPath = "/services/scipion/data/downloads/scipion/software/em/"
+    remoteFn = "xmipp_models_%s.tgz" % modelName
+    remotePath = os.path.join(depositionPath, remoteFn)
+    uploadOk = runJob("scp %s.tgz %s:%s" % (modelName, login, remotePath))
+
+    if uploadOk:
+        print(" Upload finished")
+        runJob("rm %s.tgz" % modelName)
+        print("Adding the %s model to the models.list." % modelName)
+        with open(os.path.join("src", "xmipp", "models", "models.list"), 'a') as f:
+            f.write(modelName+'\n')
+        print("Please," + red("commit and push the model.list file ") +
+              "to the devel branch!")
+    else:
+        print(red("You have no permissions to upload data in Scipion Web:\n") +
```

Comment:
It will be better: 
red("**Upload failed! You may** have no permissions to upload data in Scipion Web:\n") +
      " Please contact to 'scipion@cnb.csic.es' to upload the model"
      " **_** placed at '%s'." % os.path.join(os.getcwd(), modelName+'.tgz'))